### ISSUE TITLE
build fail on jetson nano and ubuntu 24.04

### ISSUE THREAD

**Issue by maxvaneck** [zxing-cpp/zxing-cpp#812]
as it says in the title 
pip3 install zxing-cpp fails on both my jetson nano and desktop on ubuntu 24.04  but not on windows 10  attached is the nano log 
[log.txt](https://github.com/user-attachments/files/16360801/log.txt)

**Comment by shauryagit20** [comment]
Hey There!
I am facing the same issue, did you find any way out of it?

**Comment by maxvaneck** [comment]
unfortunately no , i really want to because i want to finish my bird detector (long story).

**Comment by maxvaneck** [comment]
at this point i'm just waiting for the next release

**Comment by shauryagit20** [comment]
so I did find a way through this... not optimal but basically install the
inference package  without any dependencies - pip install --no-deps
inference-gpu and then install all the other dependencies except of
zxing.cpp

**Comment by allyssonmast** [comment]
up

**Comment by axxel** [comment]
Sorry for the late reply. Your log files does not contain any useful information about why the build failed. Can you increase the verbosity level to fix this?

**Comment by axxel** [comment]
Can anyone of you guys that ran into this provide more information what exactly is going wrong here? If not, I'm going to close this issue.

**Comment by maxvaneck** [comment]
i'll get right on it

**Comment by maxvaneck** [comment]
sorry for the late reply.i'm doing my graduation internship and tiime kind of got away from me . this is with pip3 install zxing-cpp --verbose 
[zxing.txt](https://github.com/user-attachments/files/17666629/zxing.txt)

**Comment by maxvaneck** [comment]
aha i'll see if i can upgrade that. i just assumed pip3 install would work as it does most of te time

**Comment by axxel** [comment]
Any luck with updating your compiler?

**Comment by maxvaneck** [comment]
it took quite a lot of fiddling around i installed gcc 11 as shown below but it still does not work. see the attached verbose log 

Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/aarch64-linux-gnu/11/lto-wrapper
Target: aarch64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubunt
[zxing.txt](https://github.com/user-attachments/files/17733127/zxing.txt)
u 11.4.0-2ubuntu1~20.04' --with-bugurl=file:///usr/share/doc/gcc-11/README.Bugs --enable-languages=c,ada,c++,go,d,fortran,objc,obj-c++,m2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-11 --program-prefix=aarch64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-bootstrap --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-libquadmath --disable-libquadmath-support --enable-plugin --enable-default-pie --with-system-zlib --enable-libphobos-checking=release --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --enable-fix-cortex-a53-843419 --disable-werror --enable-checking=release --build=aarch64-linux-gnu --host=aarch64-linux-gnu --target=aarch64-linux-gnu --with-build-config=bootstrap-lto-lean --enable-link-serialization=2
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 11.4.0 (Ubuntu 11.4.0-2ubuntu1~20.04)

**Comment by axxel** [comment]
The log states that cmake found a gcc-11 but no g++ compiler at all. Since this is a C++ library, that won't do ;).

**Comment by maxvaneck** [comment]
yeah.. one does tend to need a compiler to compile. i'll get right to testing once i'm done with work

**Comment by maxvaneck** [comment]
i can report success now . problem solved. tldr: install gcc 11 and g++
